### PR TITLE
fix(organization-service): add check that organization is not empty

### PR DESCRIPTION
**Pull request by daniloff200**
## Description
Add check, to restrict creating empty organizations with http request

Fixes #https://github.com/appsmithorg/appsmith/issues/774 (issue)

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/OrganizationServiceImpl.java [hunk 1]
```diff
@@ -116,6 +116,9 @@ public OrganizationServiceImpl(Scheduler scheduler,
      */
     @Override
     public Mono<Organization> createPersonal(final Organization organization, User user) {
+        if (organization.getName().length() == 0) {
```
Comment (pc9795) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/OrganizationServiceImpl.java:119]: IMO we shouldn't be handling validation logic in the service layer. We can do away with this issue by adding an `@NotEmpty` annotation in the `Organization` object's name field. 
Even for some weird reason if we are handling it here at-lease we can trim the name first (it is extreme).

**Comment by mohanarpit** [comment]
@daniloff200 Are you still working on this PR? I'm asking because I haven't seen any activity in the past few days. 

Can you please resolve the review comments and update this PR?

**Comment by daniloff200** [comment]
@mohanarpit  yeah, I will do that at the weekend.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/OrganizationServiceImpl.java (hunk 1 above)
Comment (daniloff200) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/OrganizationServiceImpl.java]: Done

**Comment by daniloff200** [comment]
@mohanarpit  done :)

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Organization.java [hunk 2]
```diff
@@ -22,6 +23,7 @@
 
     private String domain;
 
+    @NotEmpty
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Organization.java:26]:
```suggestion
    @NotEmpty(message = "Name must not be empty")
    @NotBlank(message = "Name must not be blank")
```

Comment (daniloff200) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Organization.java]: thanks for that suggestion, I've added it :)

**Review by trishaanand — APPROVED** [review]

**Pull request merged**